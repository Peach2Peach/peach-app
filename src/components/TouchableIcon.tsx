import { ColorValue, TouchableOpacity, TouchableOpacityProps } from 'react-native'
import { IconType } from '../assets/icons'
import tw from '../styles/tailwind'
import { Icon } from './Icon'

type Props = {
  id: IconType
  iconColor?: ColorValue | undefined
  iconSize?: number
} & TouchableOpacityProps

const DEFAULT_ICON_SIZE = 24
export function TouchableIcon ({ id, iconColor = tw.color('primary-main'), iconSize, ...touchableProps }: Props) {
  return (
    <TouchableOpacity {...touchableProps}>
      <Icon id={id} size={iconSize ?? DEFAULT_ICON_SIZE} color={iconColor} />
    </TouchableOpacity>
  )
}
