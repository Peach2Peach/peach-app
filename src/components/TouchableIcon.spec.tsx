import { toMatchDiffSnapshot } from 'snapshot-diff'
import { render } from 'test-utils'
import { TouchableIcon } from './TouchableIcon'
expect.extend({ toMatchDiffSnapshot })

describe('TouchableIcon', () => {
  const defaultIcon = <TouchableIcon id="chevronLeft" />
  it('should render correctly', () => {
    const { toJSON } = render(defaultIcon)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render correctly with a custom icon color and size', () => {
    const { toJSON } = render(<TouchableIcon id="chevronLeft" iconColor="#000" iconSize={32} />)

    expect(render(defaultIcon).toJSON()).toMatchDiffSnapshot(toJSON())
  })
})
