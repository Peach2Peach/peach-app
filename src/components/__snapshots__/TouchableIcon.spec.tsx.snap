// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableIcon should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <test-file-stub
    fill="#F56522"
    style={
      [
        {
          "height": 24,
          "width": 24,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`TouchableIcon should render correctly with a custom icon color 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -31,11 +31,11 @@
        "opacity": 1,
      }
    }
  >
    <test-file-stub
-     fill="#F56522"
+     fill="#000"
      style={
        Array [
          Object {
            "height": 24,
            "width": 24,"
`;

exports[`TouchableIcon should render correctly with a custom icon color and size 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -31,16 +31,16 @@
        "opacity": 1,
      }
    }
  >
    <test-file-stub
-     fill="#F56522"
+     fill="#000"
      style={
        Array [
          Object {
-           "height": 24,
-           "width": 24,
+           "height": 32,
+           "width": 32,
          },
          undefined,
        ]
      }
    />"
`;
